### PR TITLE
Refactor oic auth logic

### PR DESCRIPTION
**Pull request by jonandernovella**
The Elixir auth logic has been refactored in order to use the oic lib, and also too be able to properly fetch visas from Elixir AAI.

### CODE REVIEW

File: backend/elixir_authenticator.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 from requests.auth import HTTPBasicAuth
 import logging
-from route import app
-from flask_pyoidc import OIDCAuthentication
-
+from oic import rndstr
+from oic.oic import Client, RegistrationResponse, AuthorizationResponse, AuthorizationErrorResponse, AccessTokenResponse
+from oic.oic.message import ProviderConfigurationResponse
+from oic.oic import Grant
+from oic.oic import Token
```
Comment (viklund) [inline backend/elixir_authenticator.py]: These two should be imported on line 7.

File: backend/elixir_blueprint.py [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
+    req = flask.request.args
+    state = flask.session.get("state", None)
+    logging.debug("Request coming")
+    logging.debug(req)
+    if 'error' in req:
+        logging.debug("Response contains error")
+        return jsonify({"status": "Error"}), 401
+    elif 'code' in req and state is not None:
+        logging.debug("Code resp")
+        elixir_authenticator.handle_authentication_response()
```
Comment (viklund) [inline backend/elixir_blueprint.py]: You're using the `elixir_authenticator` module as if it was an object already. Just remove all the globals from `elixir_authenticator.py` and instantiate the `ElixirAuthenticator` at the top of this function.

Comment (jonandernovella) [reply backend/elixir_blueprint.py]: This is fixed in the next PR, as I encapsulated these classes when building the test suite.

Comment (viklund) [reply backend/elixir_blueprint.py]: Well, if the test PR reoganizes everything I don't really see the point of this one? Maybe we should just take the other one and split it into the reorg and a test one?

Comment (jonandernovella) [reply backend/elixir_blueprint.py]: Done

**Pull request merged**